### PR TITLE
 Student Feedback Question illigal symbols.

### PR DESCRIPTION
**Pull request by a22karja**
In a courses listentries the input field Student Feedback Question: now have the same rules as the name. 
This should stop any bad symbols from entering  Student Feedback Question:

For testing:
Try to save symbols in Student Feedback Question: field, that could crash either the database or website:
image

### CODE REVIEW

File: DuggaSys/sectioned.js [hunk 1]
```diff
@@ -3735,7 +3738,12 @@ function quickValidateForm(formid, submitButton) {
     var deadlinepart = document.getElementById('inputwrapper-deadline');
     var deadlinedisplayattribute = deadlinepart.style.display;
     valid = true;
-    valid &= validateSectName('sectionname');
+    valid &= validateSectName('sectionname','dialog10');
+    console.log('a22karja after name');
+    console.log(valid);
+    valid &= validateSectName('fdbckque','dialog11');
+    console.log('a22karja after question');
+    console.log(valid);
```
Comment (b22marah) [inline DuggaSys/sectioned.js:3746]: Remove console.logs

File: DuggaSys/sectioned.js [hunk 2]
```diff
@@ -331,6 +331,9 @@ function selectItem(lid, entryname, kind, evisible, elink, moment, gradesys, hig
   // Set Lid
   $("#lid").val(lid);
 
+  validateSectName('sectionname','dialog10');
+  validateSectName('fdbckque','dialog11');
```
Comment (b22marah) [inline DuggaSys/sectioned.js:335]: fdbckque is a little ambigious, change to clearer name like feedbackQue.

Comment (b22marah) [inline DuggaSys/sectioned.js:335]: Could also change function name to validateSectionName for clearity

File: DuggaSys/sectioned.php [hunk 3]
```diff
@@ -323,7 +324,14 @@ function updateGithubDir($pdo, $githubDir, $lid) {
 					<div id='inputwrapper-visibility' class='inputwrapper'><span>Visibility:</span><select style='align:right;' id='visib'></select></div>
 					<div id='inputwrapper-group' class='inputwrapper'><span>Group type:</span><select style='align:right;' id='grptype'></select></div>
 					<div id='inputwrapper-Feedback' class='inputwrapper'><span>Enable Student Feedback:</span><input type="checkbox"  style='align:center;' id='fdbck' title='Student feedback checkbox' onchange='showFeedbackquestion()'></input></div>
-					<div id='inputwrapper-FeedbackQuestion' class='inputwrapper' style='display:none;'><span>Student Feedback Question:</span><input type="input"  class='textinput' id='fdbckque' value='How would you grade the dugga?'></input></div>
+					<!-- a22karja -->
```
Comment (b22marah) [inline DuggaSys/sectioned.php:327]: remove

File: DuggaSys/sectioned.php [hunk 4]
```diff
@@ -323,7 +324,14 @@ function updateGithubDir($pdo, $githubDir, $lid) {
...
+					<div id='inputwrapper-FeedbackQuestion' class='inputwrapper' style='display:none;'>
```
Comment (b22marah) [inline DuggaSys/sectioned.php:328]: Remove in-line css

File: DuggaSys/sectioned.php [hunk 5]
```diff
@@ -323,7 +324,14 @@ function updateGithubDir($pdo, $githubDir, $lid) {
...
+					<span>Student Feedback Question:</span>
+					<input type="input" onkeyup="quickValidateForm('editSection', 'saveBtn');" onchange="validateSectName('fdbckque','dialog11')" class='textinput' id='fdbckque' value='How would you grade the dugga' ></input>
+					</div>
+					<div class="formDialog" style="display: block;  margin-left:40px;">
```
Comment (b22marah) [inline DuggaSys/sectioned.php:332]: remove inline css

File: DuggaSys/sectioned.php [hunk 6]
```diff
@@ -323,7 +324,14 @@ function updateGithubDir($pdo, $githubDir, $lid) {
...
+  		      			<span id="dialog11" style="display: none; left:0px;" class="formDialogText">Illegal characters found in the title!<br>Valid characters: A-Ö, 0-9.</span>
```
Comment (b22marah) [inline DuggaSys/sectioned.php:333]: remove inline css

**Review by b22marah — CHANGES_REQUESTED** [review]
Besides these changes everything looks good!

**Review by b22marah — CHANGES_REQUESTED** [review]
Still haven't removed inline css, this needs to be fixed for me to approve this PR

**Review by b22marah — APPROVED** [review]
LGTM!

**Review by a22moamy — CHANGES_REQUESTED** [review]
If you enter feedback in the input, then uncheck the option to enable student feedback, then you won't be able to save the form even if all other inputs are good:
Skärmavbild 2024-05-13 kl  11 26 39

You have to then manually go out of the form and go into it again to save it, this is a bit tedious though. Please fix this and everything else should be good to go :)

**Review by a22moamy — APPROVED** [review]
Looks great! Good job!

**Pull request merged**